packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -135,49 +157,43 @@ export const disconnectProvider = () => async (dispatch: Dispatch, getState: Get
         type: METADATA.SET_PROVIDER,
         payload: undefined,
     });
+
+    // dispose metadata values (not keys)
     dispatch(disposeMetadata());
 };
 
 const handleProviderError = (error: Error) => (dispatch: Dispatch, getState: GetState) => {
     const { metadata } = getState();
-    const unexpectedError = 'labeling action failed with unexpected error';
-    if (!error) {
-        return dispatch(
-            notificationActions.addToast({
-                type: 'error',
-                error: unexpectedError,
-            }),
-        );
-    }
-    // @ts-ignore todo: type error;
-    switch (error.status) {
-        case 403:
-        case 401:
-            if (getState().metadata.provider) {
-                dispatch(
-                    notificationActions.addToast({
-                        type: 'error',
-                        error: `Failed to sync labeling data with cloud provider (${
-                            metadata!.provider!.type
-                        }). User was logged out.`,
-                    }),
-                );
-                dispatch(disconnectProvider());
-            }
-            break;
-        case 404:
-            dispatch(
-                notificationActions.addToast({
-                    type: 'error',
-                    error: 'Failed to find metadata in cloud provider.',
-                }),
-            );
-            dispatch(disconnectProvider());
-            break;
-        default:
-            dispatch(notificationActions.addToast({ type: 'error', error: unexpectedError }));
-            break;
+    let errorMsg = 'labeling action failed with unexpected error';
+    if (error) {
+        // @ts-ignore todo: type error;
```

Comment:
We should try to minimize the use of `ts-ignore` unless it's really necessary.

Reply:
Yes. I am definitely about to do it. But was not planning to do it in this PR - this ts-ignore was not actually introduced by this PR, you only see it in diff because I reorganized this section of code.
I am planning to rework error handling as part of following PR which will revisit provider logic and authentication and most like will introduce official google lib instead of plain rest api implementation - and this might affect how errors are to be handled.